src/main/java/com/blackduck/integration/polaris/common/cli/PolarisDownloadUtility.java

```diff
@@ -173,16 +173,20 @@ public File getOrCreateVersionFile() throws IOException {
         return versionFile;
     }
 
-    public String getDownloadUrlFormat() {
+    public String getDownloadUrlFormat() throws IntegrationException {
         if (OperatingSystemType.MAC == operatingSystemType) {
             // If the OS Architecture is Mac non-ARM, the return tool name as "%s_cli-macosx.zip"
             // If the OS Architecture is Mac ARM architecture, the return tool name as "%s_cli-macos_arm.zip"
             FilePath workspace = new FilePath(new File(installDirectory.getPath()));
             String arch = getAgentOsArch(workspace);
-            if (arch.startsWith("arm") || arch.startsWith("aarch")) {
-                return polarisServerUrl + PolarisDownloadUtility.MAC_ARM_DOWNLOAD_URL_FORMAT;
+            if (arch != null) {
+                if (arch.startsWith("arm") || arch.startsWith("aarch")) {
+                    return polarisServerUrl + PolarisDownloadUtility.MAC_ARM_DOWNLOAD_URL_FORMAT;
+                } else {
+                    return polarisServerUrl + PolarisDownloadUtility.MAC_DOWNLOAD_URL_FORMAT;
+                }
             } else {
-                return polarisServerUrl + PolarisDownloadUtility.MAC_DOWNLOAD_URL_FORMAT;
+                throw new IntegrationException("OS architecture of MAC could not be determined. 'arch' is null.");
```

Comment:
@zaman-akib Do we really want to update this existing logic? I'm a bit doubtful about the impact.

Reply:
@blackduck-saraf In SonarCloud, it is showing the `arch` can be `null` so we should handle that. So should we only log this as warning or throw exception like this?

Reply:
Okay, got it! Let's discuss this in today's sync up.

Reply:
As discussed, if we don't fix this, it will show in the bugs under sonarcloud results. Also, we don't see a possibility of getting `null` arch in any value case. So, I think it's fine to throw the exception. Thanks!